### PR TITLE
Use alarm() for interval checkpointing.

### PR DESCRIPTION
**Pull request by karya0**
This replaces the existing somewhat complex code with (hopefully) a much simpler and less error prone one.

### CODE REVIEW

**Comment by karya0** [comment]
@rohgarg: @gc00: Can you take a look at this one?

File: src/dmtcp_coordinator.cpp [hunk 1]
```diff
@@ -1554,18 +1556,26 @@ void DmtcpCoordinator::writeRestartScript()
...
 }
 
-static void SIGINTHandler(int signum)
+static void signalHandler(int signum)
 {
-  prog.handleUserCommand('q');
+  if (signum == SIGINT) {
+    prog.handleUserCommand('q');
+  } else if (signum == SIGALRM) {
+    timerExpired = 1;
```
Comment (gc00) [inline src/dmtcp_coordinator.cpp]: Shouldn't this be:  `timerExpired = true;`?  (Too much thinking in C. :-) )

**Comment by gc00** [comment]
I had one minor "line note".  Once that is fixed, go ahead and push it in.

(I like it.  It's definitely more straightforward.)

**Comment by gc00** [comment]
@karya0, Before you push it in, could you do:  `grep onTimeoutInterval  src/*.cpp`.  There are some comments in `src/dmtcp_coordinator.cpp` that need to be updated.  (And please do: `git pull --rebase` before the `grep`.  I created an accidental conflict (long story), and you'll find another reference to `onTimeoutInterval` there.

**Pull request closed without merging**